lib/gooddata/rest/connection.rb

```diff
@@ -288,14 +291,16 @@ def download(what, where, options = {})
             :verify_ssl => verify_ssl
           }
           RestClient::Request.execute(raw) do |chunk, _x, response|
-            if response.code.to_s != '200'
+            if response.code.to_s == '202'
+              fail RestRetry, 'Got 202, retry'
```

Comment:
Please consider refactoring this. Flow control should be implemented using `throw/catch`, not exceptions.